mac/hid.c

```diff
@@ -513,12 +513,7 @@ static struct hid_device_info *create_device_info_with_usage(IOHIDDeviceRef dev,
 		   9+1+20+1=31 bytes buffer, but allocate 32 for simple alignment */
 		cur_dev->path = calloc(1, 32);
 		if (cur_dev->path != NULL) {
-			/* Yes, compiler, we know that snprintf is preferable,
-			   but we're not ready to abandon older macOS-es/SDKs where it is not yet available */
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
-			sprintf(cur_dev->path, "DevSrvsID:%llu", entry_id);
-#pragma GCC diagnostic pop
+			snprintf(cur_dev->path, 32, "DevSrvsID:%llu", entry_id);
```

Comment:
I'd prefer if we had a local named constant for the `32` value

Reply:
I had followed what you already had. Invented a `const size_t path_len` for it now.

Reply:
true, I tend to be lazy about named constants, but usually not when it comes to multiple usages of such